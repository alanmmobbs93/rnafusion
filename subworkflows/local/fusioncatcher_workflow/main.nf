include { FUSIONCATCHER } from '../../../modules/local/fusioncatcher/detect/main'

workflow FUSIONCATCHER_WORKFLOW {
    take:
        reads                   // channel [ meta, [ fastqs ] ]
        fusioncatcher_ref       // channel [ meta, path       ]
        run_fusioncatcher       // boolean
        all                     // boolean
        fusioninspector_only    // boolean
        fusioncatcher_fusions   // path, string

    main:
        ch_versions   = Channel.empty()
        ch_dummy_file = file("$baseDir/assets/dummy_file_fusioncatcher.txt", checkIfExists: true)

        if (( run_fusioncatcher || all) && !fusioninspector_only ) {
            if (fusioncatcher_fusions){

                ch_fusioncatcher_fusions = reads.combine(Channel.value(file(fusioncatcher_fusions, checkIfExists:true)))
                                            .map { meta, reads, fusions -> [ meta, fusions ] }
            } else {

                FUSIONCATCHER (
                    reads,
                    fusioncatcher_ref
                )
                ch_fusioncatcher_fusions = FUSIONCATCHER.out.fusions
                ch_versions = ch_versions.mix(FUSIONCATCHER.out.versions)
            }
        }
        else {
            ch_fusioncatcher_fusions = reads.combine(Channel.value(file(ch_dummy_file, checkIfExists:true)))
                                        .map { meta, reads, fusions -> [ meta, fusions ] }
        }

    emit:
        fusions  = ch_fusioncatcher_fusions
        versions = ch_versions
    }

