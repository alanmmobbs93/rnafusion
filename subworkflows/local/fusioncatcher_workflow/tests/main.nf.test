nextflow_workflow {

    name "Test Subworkflow FUSIONCATCHER_WORKFLOW"
    script "../main.nf"
    workflow "FUSIONCATCHER_WORKFLOW"
    tag "subworkflow"
    tag "fusioncatcher"
    tag "fusioncatcher/download"
    tag "fusioncatcher/detect"

    // Test
    test("FUSIONCATCHER_WORKFLOW - Homo sapiens - FASTQs chr4") {

        /*setup {
            // Download reference files for fusioncatch
            run("FUSIONCATCHER_DOWNLOAD") {
                script "../../../../modules/local/fusioncatcher/download/main.nf"
                process {
                """
                // No arguments required. Human version v102
                """
                }
            }
        }*/

        when {
            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [ id: "test_fastqs", single_end: false ],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_1.fq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_2.fq.gz", checkIfExists: true)
                        ]
                    ] )

                // ch_references
                input[1] = Channel.fromPath("/scratch/moebbs/NEXTFLOW_CODE/reference_rnafusion/human_v102").map{ [ [id: "human_v102"], it] } //FUSIONCATCHER_DOWNLOAD.out.reference

                // fusioncatcher (boolean)
                input[2] = true

                // all (boolean)
                input[3] = true

                // fusioninspector_only (boolean)
                input[4] = false

                // fusioncatcher_fusions (string path)
                input[5] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

}
