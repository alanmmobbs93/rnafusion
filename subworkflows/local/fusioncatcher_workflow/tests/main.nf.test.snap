{
    "FUSIONCATCHER_WORKFLOW - Homo sapiens - FASTQs chr4": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_fastqs",
                            "single_end": false
                        },
                        "test_fastqs.fusioncatcher.fusion-genes.txt:md5,c826a24c49abfcec8164c478e1e74892"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6462518e02774293fa221cead8373305"
                ],
                "fusions": [
                    [
                        {
                            "id": "test_fastqs",
                            "single_end": false
                        },
                        "test_fastqs.fusioncatcher.fusion-genes.txt:md5,c826a24c49abfcec8164c478e1e74892"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6462518e02774293fa221cead8373305"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-18T21:26:47.477736"
    }
}